yaml/parse.ts

```diff
@@ -5,9 +5,31 @@
 // This module is browser compatible.
 
 import { type CbFunction, load, loadAll } from "./_loader/loader.ts";
-import type { LoaderStateOptions } from "./_loader/loader_state.ts";
+import { replaceSchemaNameWithSchemaClass } from "./mod.ts";
 
-export type ParseOptions = LoaderStateOptions;
+/**
+ * Options for parsing YAML.
+ */
+export interface ParseOptions {
+  /** Uses legacy mode */
+  legacy?: boolean;
+  /** The listener */
+  // deno-lint-ignore no-explicit-any
+  listener?: ((...args: any[]) => void) | null;
+  /** string to be used as a file path in error/warning messages. */
+  filename?: string;
+  /**
+   * Specifies a schema to use.
+   *
+   * Schema class or its name.
+   */
+  // deno-lint-ignore no-explicit-any
+  schema?: any;
```

Comment:
I don't see a good reason as to why this should be `any`.
```suggestion
  schema?: string | Schema;
```